app/javascript/components/track/IterationSummary.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState, useRef } from 'react'
+import { TestRunStatus } from '../student/Editor'
+import { fromNow } from '../../utils/time'
+import { SubmissionMethodIcon } from './iteration-summary/SubmissionMethodIcon'
+import { TestsStatusSummary } from './iteration-summary/TestsStatusSummary'
+import { AnalysisStatusSummary } from './iteration-summary/AnalysisStatusSummary'
+import { IterationsChannel } from '../../channels/iterationsChannel'
+
+export type Iteration = {
+  id: number
+  idx: number
+  submissionMethod: SubmissionMethod
+  createdAt: Date
+  testsStatus: TestRunStatus
+  representationStatus: RepresentationStatus
+  analysisStatus: AnalysisStatus
+}
+
+export enum SubmissionMethod {
+  CLI = 'cli',
+  API = 'api',
+}
+
+export enum RepresentationStatus {
+  NOT_QUEUED = 'not_queued',
+  QUEUED = 'queued',
+  APPROVED = 'approved',
+  DISAPPROVED = 'disapproved',
+  INCONCLUSIVE = 'inconclusive',
+  EXCEPTIONED = 'exceptioned',
+  CANCELLED = 'cancelled',
+}
+
+export enum AnalysisStatus {
+  NOT_QUEUED = 'not_queued',
+  QUEUED = 'queued',
+  APPROVED = 'approved',
+  DISAPPROVED = 'disapproved',
+  INCONCLUSIVE = 'inconclusive',
+  EXCEPTIONED = 'exceptioned',
+  CANCELLED = 'cancelled',
+}
+
+export function IterationSummary(props: { iteration: Iteration }) {
+  const [iteration, setIteration] = useState(props.iteration)
+  const submissionMethodLabels = {
+    [SubmissionMethod.CLI]: 'CLI',
+    [SubmissionMethod.API]: 'API',
+  }
+  const channel = useRef<IterationsChannel | undefined>()
+
+  useEffect(() => {
+    channel.current = new IterationsChannel(iteration, (iteration) => {
+      setIteration(iteration)
+    })
+
+    return () => {
+      channel.current?.disconnect()
+    }
+  }, [iteration])
+
+  useEffect(() => {
+    return () => {
+      channel.current?.disconnect()
+    }
+  }, [channel])
+
+  return (
+    <div className="header">
```

Comment:
```suggestion
    <header className="header">
```

If the classname is header, it's probably a `<header>`

Reply:
@kntsoriano This actually shouldn't be in a header at all. It will sit within the header (where we should update to what DJ says, but this have the top-level component class).

```suggestion
    <div className="c-iteration-summary">
```